test/e2e/framework/console-operator.go

```diff
@@ -0,0 +1,390 @@
+package framework
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	consoleapi "github.com/openshift/console-operator/pkg/api"
+)
+
+// set the operator config to a pristine state to start a next round of tests
+// this should by default nullify out any customizations a user sets
+func Pristine(t *testing.T, client *ClientSet) (*operatorsv1.Console, error) {
+	t.Helper()
+	operatorConfig, err := client.Operator.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	copy := operatorConfig.DeepCopy()
+	cleanSpec := operatorsv1.ConsoleSpec{}
+	// we can set a default management state & log level, but
+	// nothing else should be necessary
+	cleanSpec.ManagementState = operatorsv1.Managed
+	cleanSpec.LogLevel = operatorsv1.Normal
+	copy.Spec = cleanSpec
+	return client.Operator.Consoles().Update(copy)
+}
+
+func MustPristine(t *testing.T, client *ClientSet) *operatorsv1.Console {
+	t.Helper()
+	operatorConfig, err := Pristine(t, client)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return operatorConfig
+}
+
+func isOperatorManaged(cr *operatorsv1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1.Managed
+}
+
+func isOperatorUnmanaged(cr *operatorsv1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1.Unmanaged
+}
+
+func isOperatorRemoved(cr *operatorsv1.Console) bool {
+	return cr.Spec.ManagementState == operatorsv1.Removed
+}
+
+type operatorStateReactionFn func(cr *operatorsv1.Console) bool
+
+func ensureConsoleIsInDesiredState(t *testing.T, client *ClientSet, state operatorsv1.ManagementState) error {
+	t.Helper()
+	var operatorConfig *operatorsv1.Console
+	// var checkFunc func()
+	var checkFunc operatorStateReactionFn
+
+	switch state {
+	case operatorsv1.Managed:
+		checkFunc = isOperatorManaged
+	case operatorsv1.Unmanaged:
+		checkFunc = isOperatorUnmanaged
+	case operatorsv1.Removed:
+		checkFunc = isOperatorRemoved
+	}
+
+	err := wait.Poll(1*time.Second, AsyncOperationTimeout, func() (stop bool, err error) {
+		operatorConfig, err = client.Operator.Consoles().Get(consoleapi.ConfigResourceName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return checkFunc(operatorConfig), nil
+	})
+	if err != nil {
+		DumpObject(t, "the latest observed state of the console resource", operatorConfig)
+		DumpOperatorLogs(t, client)
+		return fmt.Errorf("failed to wait to change console operator state to 'Removed': %s", err)
+	}
+	return nil
+}
+
+func ManageConsole(t *testing.T, client *ClientSet) error {
```

Comment:
Not necessary to export, since it's only used in the `MustManageConsole()`

Reply:
will do!